src/point.rs

```diff
@@ -0,0 +1,101 @@
+//! A 2d point.
+
+use std::fmt;
+use std::ops::{Add, AddAssign, Sub, SubAssign};
+
+use crate::Vec2;
+
+/// A 2d point.
+#[derive(Clone, Copy, Default, PartialEq)]
+pub struct Point {
+    /// The x coordinate.
+    pub x: f64,
+    /// The y coordinate.
+    pub y: f64,
+}
+
+impl Point {
+    /// The point (0, 0).
+    pub const ZERO: Point = Point::new(0., 0.);
```

Comment:
A case can be made that this should be called ORIGIN instead of ZERO.

Reply:
Here and with `Size` names were in deference to existing APIs. (I'd wanted to call the `(0, 0)` size 'empty', but landed on zero after looking around.)